spec/unit/transaction_spec.rb

```diff
@@ -600,6 +600,18 @@ def transaction_with_resource(resource)
       transaction.prefetch_if_necessary(resource)
     end
 
+    it "should not catch SystemExit without future_features flag" do
+      Puppet.settings[:future_features] = false
+      resource.provider.class.expects(:prefetch).raises(SystemExit, "message")
+      expect { transaction.prefetch_if_necessary(resource) }.to raise_error(SystemExit)
```

Comment:
Puppet has an `exit_with` matcher for this, eg `expect { ... }.to exit_with(0)`

Reply:
I am actually trying to test that it does not rescue this exception, as an example.  Is this exit_with(0) more what you intended from your comment on the ticket?

Reply:
Right, it's just that `exit_with` is a short hand for what you have. But either are fine since you're not trying to assert which code we're exiting with.